src/bun.js/bindings/CommonJSModuleRecord.h

```diff
@@ -44,6 +44,17 @@ class JSCommonJSModule final : public JSC::JSDestructibleObject {
     mutable JSC::WriteBarrier<JSString> m_dirname;
     // Initialized lazily; can be overridden.
     mutable JSC::WriteBarrier<Unknown> m_paths;
+    // Children must always be tracked in case the script decides to access
+    // `module.children`. In that case, all children of the children may also
+    // need their children fields to exist. To avoid allocating *JSArray for
+    // each module, the children array is constructed internally as a HashSet
+    // of pointers. If accessed, the HashSet will still be maintained to detect
+    // when new items are required. The WriteBarrier+HashSet adds 32 bytes to
+    // JSCommonJSModule. `m_childrenValue` can be set to any value via the
+    // user-exposed setter, but Bun does not test that behavior besides
+    // ensuring it does not crash.
+    mutable JSC::WriteBarrier<Unknown> m_childrenValue;
+    WTF::HashSet<JSCommonJSModule*> m_children;
```

Comment:
Can we use a WTF::FixedVector? Hash table will be more expensive

Reply:
i chose hash table because it has to deduplicate entries. so instead it'll have to iterate through the list to find a duplicate. this might be fine in the end.

alternatively we can deduplicate when .children is accessed; use the assumption that people require things only once.